File: python/bob/bob.py
def response(hey_bob):
    bob_response = "Whatever."
    if (not hey_bob or hey_bob.strip() == "" or len(hey_bob) == 0):
        bob_response = "Fine. Be that way!"
    elif (hey_bob.endswith("?") and hey_bob.isupper()):
        bob_response = "Calm down, I know what I'm doing!"
    elif hey_bob.strip().endswith("?"):
        bob_response = "Sure."
    elif hey_bob.isupper() :
        bob_response = "Whoa, chill out!"
    return bob_response

File: zig/bob/bob.zig
const std = @import("std");
const ascii = std.ascii;

pub fn response(s: []const u8) []const u8 {
    const hey_bob = std.mem.trim(u8, s, &ascii.whitespace);

    if (hey_bob.len == 0) return "Fine. Be that way!";

    const is_shout = isallcaps(hey_bob);
    const is_question = hey_bob[hey_bob.len - 1] == '?';

    if (is_shout and is_question) return "Calm down, I know what I'm doing!";
    if (is_shout) return "Whoa, chill out!";
    if (is_question) return "Sure.";
    return "Whatever.";
}

pub fn isallcaps(str: []const u8) bool {
    var e_upper: bool = false;
    for (str) |letra| {
        if (ascii.isLower(letra)) return false;
        if (ascii.isUpper(letra)) e_upper = true;
    }
    return e_upper;
}
